### PR TITLE
Proxy support for SourceAddressAdapter

### PR DESCRIPTION
**Pull request by dys**
Using proxy_manager_for to pass source_address to the ProxyManager.

### CODE REVIEW

File: requests_toolbelt/adapters/source.py [hunk 1]
```diff
@@ -60,3 +60,7 @@ def init_poolmanager(self, connections, maxsize, block=False):
             maxsize=maxsize,
             block=block,
             source_address=self.source_address)
+
+    def proxy_manager_for(self, *args, **kwargs):
+        kwargs['source_address'] = self.source_address
```
Comment (sigmavirus24) [inline requests_toolbelt/adapters/source.py]: What if instead we do

```py
kwargs.setdefault('source_address', self.source_address)
```

I suppose it's possible that's already being sent and we wouldn't want to forcibly override it? I'm not sure.

**Comment by sigmavirus24** [comment]
Also, the tests are failing on style compliance: https://travis-ci.org/sigmavirus24/requests-toolbelt/jobs/143071491#L165

**Comment by dys** [comment]
Ha! 3 characters over. Doh. The class name is longer than SSLAdapter. I'll see if I can find a way to make it a bit shorter.

RE: setdefault, SSLAdapter is doing something similar with ssl_version. Would it be good to change there too for consistency?

File: requests_toolbelt/adapters/source.py [hunk 2]
```diff
@@ -60,3 +60,7 @@ def init_poolmanager(self, connections, maxsize, block=False):
...
+        return super(SourceAddressAdapter, self).proxy_manager_for(*args, **kwargs)
```
Comment (sigmavirus24) [inline requests_toolbelt/adapters/source.py]: Try:

```py
return super(SourceAddressAdapter, self).proxy_manager_for(
    *args, **kwargs
)
```

**Comment by sigmavirus24** [comment]
> RE: setdefault, SSLAdapter is doing something similar with ssl_version. Would it be good to change there too for consistency?

So that should be a different pull request if we decide to do it at all. I'd like @Lukasa to weigh in before deciding to do this or not.

**Comment by sigmavirus24** [comment]
Fair enough. @dys if you just fix up the style issue, I'm happy to merge this. :)

**Comment by sigmavirus24** [comment]
Thanks @dys!

**Pull request merged**